internal/user/repository/eventsourcing/eventstore_test.go

```diff
@@ -0,0 +1,2317 @@
+				user: &model.User{
+					ObjectRoot: es_models.ObjectRoot{Sequence: 1},
+					Profile: &model.Profile{
+						UserName:  "UserName",
+						FirstName: "FirstName",
+						LastName:  "LastName",
+					},
+					Email: &model.Email{
+						EmailAddress:    "EmailAddress",
+						IsEmailVerified: true,
+					},
+				},
+			},
+			res: res{
+				user: &model.User{ObjectRoot: es_models.ObjectRoot{Sequence: 1},
+					Profile: &model.Profile{
+						UserName:  "UserName",
+						FirstName: "FirstName",
+						LastName:  "LastName",
+					},
+					Email: &model.Email{
+						EmailAddress:    "EmailAddress",
+						IsEmailVerified: true,
+					},
+				},
+			},
+		},
+		{
+			name: "email as username",
+			args: args{
+				es:  GetMockManipulateUserWithInitCodeGen(ctrl, repo_model.User{Profile: &repo_model.Profile{UserName: "EmailAddress", FirstName: "FirstName", LastName: "LastName"}, Email: &repo_model.Email{EmailAddress: "EmailAddress", IsEmailVerified: true}}),
+				ctx: auth.NewMockContext("orgID", "userID"),
+				user: &model.User{ObjectRoot: es_models.ObjectRoot{Sequence: 1},
+					Profile: &model.Profile{
+						FirstName: "FirstName",
+						LastName:  "LastName",
+					},
+					Email: &model.Email{
+						EmailAddress:    "EmailAddress",
+						IsEmailVerified: true,
+					},
+				},
+			},
+			res: res{
+				user: &model.User{ObjectRoot: es_models.ObjectRoot{Sequence: 1},
+					Profile: &model.Profile{
+						UserName:  "EmailAddress",
+						FirstName: "FirstName",
+						LastName:  "LastName",
+					},
+					Email: &model.Email{
+						EmailAddress:    "EmailAddress",
+						IsEmailVerified: true,
+					},
+				},
+			},
+		},
+		{
+			name: "with verified phone number",
+			args: args{
+				es:  GetMockManipulateUserWithInitCodeGen(ctrl, repo_model.User{Profile: &repo_model.Profile{UserName: "EmailAddress", FirstName: "FirstName", LastName: "LastName"}, Email: &repo_model.Email{EmailAddress: "EmailAddress", IsEmailVerified: true}, Phone: &repo_model.Phone{PhoneNumber: "PhoneNumber", IsPhoneVerified: true}}),
+				ctx: auth.NewMockContext("orgID", "userID"),
+				user: &model.User{ObjectRoot: es_models.ObjectRoot{Sequence: 1},
+					Profile: &model.Profile{
+						FirstName: "FirstName",
+						LastName:  "LastName",
+						UserName:  "UserName",
+					},
+					Email: &model.Email{
+						EmailAddress:    "UserName",
+						IsEmailVerified: true,
+					},
+					Phone: &model.Phone{
+						PhoneNumber:     "UserName",
+						IsPhoneVerified: true,
+					},
+				},
+			},
+			res: res{
+				user: &model.User{ObjectRoot: es_models.ObjectRoot{Sequence: 1},
+					Profile: &model.Profile{
+						UserName:  "UserName",
+						FirstName: "FirstName",
+						LastName:  "LastName",
+					},
+					Email: &model.Email{
+						EmailAddress:    "EmailAddress",
+						IsEmailVerified: true,
+					},
+					Phone: &model.Phone{
+						PhoneNumber:     "UserName",
+						IsPhoneVerified: true,
+					},
+				},
```

Comment:
for me the tests are still a bit confusing. Data passed into the mock and the test itself differs and result may differ as well, e.g. EmailAddress, PhoneNumber:
mock: EmailAddress, PhoneNumber
test: UserName, Username
result: EmailAddress, Username

tests work, because these fields don't get tested, only the necessary booleans
and it's partly ok as for the flow testing, these are the required ones
but imo we should either test these as well or leave the empty